src/trim.h

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2017 Evan Teran
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+//
+
+// String-Trimming Lambdas
+// Author: Evan Teran / evan-teran
+// URL: https://stackoverflow.com/a/217605
+// MIT License Derived from Stack Overflow Policy:
```

Comment:
It appears [I am mistaken](https://meta.stackexchange.com/questions/272956/a-new-code-license-the-mit-this-time-with-attribution-required) on [which license is required for code snippets from Stack Overflow](https://meta.stackexchange.com/a/12528). The whole discussion is rather heated and frankly, confusing, but it appears that this should be licensed [CC BY-SA 3.0](https://creativecommons.org/licenses/by-sa/3.0/) according to this post.  I believe [BSD and CC are still compatible licenses](https://opensource.stackexchange.com/questions/324/are-permissive-licenses-mit-bsd-zlib-compatible-with-cc-by), so I'll just update the license to CC.